### PR TITLE
Add noti (new formula)

### PR DESCRIPTION
**Pull request by mmizutani**
[Noti](https://github.com/variadico/noti) is a OS X/Linux terminal utility which displays a push notification message in various forms after command processes finish. This utility is built with go, and this formula allows users to install noti either using the prebuilt binary or the repository master sources. Thank you.

### CODE REVIEW

File: Library/Formula/noti.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require "language/go"
+
+class Noti < Formula
+  desc "Displays a notification after a terminal process finishes."
+  homepage "https://github.com/variadico/noti"
+  version "1.3.0"
+  head "https://github.com/variadico/noti.git"
+  if OS.mac?
+    url "https://github.com/variadico/noti/releases/download/v1.3.0/noti1.3.darwin-amd64.tar.gz"
```
Comment (mikemcquaid) [inline Library/Formula/noti.rb]: This needs to be built from source and we don't accept `OS.mac?` in Homebrew (Homebrew only supports OS X)

**Comment by mmizutani** [comment]
I have fixed it so that this utility is always built from the source.
Thank you for your review.

**Comment by strages** [comment]
when will this be merged? :)

File: Library/Formula/noti.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  depends_on "go" => :build
+
+  def install
+    mkdir_p buildpath/"src/github.com/variadico"
+    ln_s buildpath, buildpath/"src/github.com/variadico/noti"
+    ENV["GOPATH"] = buildpath
+    cd buildpath/"src/github.com/variadico/noti" do
+      if build.devel? || build.head?
```
Comment (bfontaine) [inline Library/Formula/noti.rb]: You can use `unless build.stable?` for that.

File: Library/Formula/noti.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      end
+    end
+  end
+
+  def caveats; <<-EOS.undent
+    If you want noti to send notifications to smartphones, export the environment variable
+    PUSHBULLET_ACCESS_TOKEN in your .bashrc, .zshrc, etc.
+    See https://github.com/variadico/noti#pushbullet-notifications for more details.
+    EOS
+  end
```
Comment (bfontaine) [inline Library/Formula/noti.rb]: Please remove this since it’s not specific to Homebrew.

File: Library/Formula/noti.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  test do
+    system "noti", "-t", "'Noti'", "-m", "'Noti recipe installation test has finished.'"
```
Comment (bfontaine) [inline Library/Formula/noti.rb]: Use `"#{bin}/noti"` instead of `"noti"` to ensure it uses the correct binary.

**Comment by bfontaine** [comment]
@strages We need to fix a few things then wait for the CI to build/test it.

**Comment by mmizutani** [comment]
@bfontaine I believe this pull request is finally ready.

File: Library/Formula/noti.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  def caveats; <<-EOS.undent
+    If you want noti to send notifications to smartphones, export the environment variable
+    PUSHBULLET_ACCESS_TOKEN in your .bashrc, .zshrc, etc.
+    See https://github.com/variadico/noti#pushbullet-notifications for more details.
+    EOS
+  end
+
+  test do
+    system "noti", "-t", "Noti", "-m", "'Noti recipe installation test has finished.'"
```
Comment (bfontaine) [inline Library/Formula/noti.rb]: Please use `"#{bin}/noti"` instead of `"noti"`. My previous comment was hidden by GitHub because the diff changed.

**Comment by derekperkins** [comment]
Can't wait for this!!

**Comment by bfontaine** [comment]
@mmizutani It’s all good except that you need to remove caveats: we use them for things that are specific to Homebrew; general instructions should go on the project homepage. Thanks!

**Comment by apjanke** [comment]
Looks good now. Could you squash this to a single commit, so we end up with a tidy commit history?

**Comment by mmizutani** [comment]
Squashing done. Thanks.

**Comment by apjanke** [comment]
All good now. Merged. Thank you for your contribution to Homebrew!

Minor style nit for future reference: our preferred format for commit & PR titles for new formulae is "noti 1.3.0 (new formula)".

**Comment by mmizutani** [comment]
@apjanke @bfontaine Thank you.

**Pull request closed without merging**